catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -255,6 +271,64 @@ private void log(QueryResponse queryResponse) {
         }
       }
     }
+    if (LOGGER.isDebugEnabled()) {
+      Map<String, Serializable> req_metrics = queryResponse.getRequest().getProperties();
+      Map<String, Serializable> resp_metrics = queryResponse.getProperties();
+      // Combine the request and response metrics to log
+      Map<String, Serializable> all_metrics = new HashMap<>();
+      all_metrics.putAll(filterMetrics(req_metrics, QMB));
+      all_metrics.putAll(filterMetrics(resp_metrics, QMB));
+
+      LOGGER.debug("QueryMetrics: {}", serializeMetrics(all_metrics, QMB));
+    }
+  }
+
+  private Map<String, Serializable> filterMetrics(Map<String, Serializable> src, String base) {
+    Map<String, Serializable> filtered =
+        src.entrySet().stream()
+            .filter(x -> x.getKey().startsWith(base))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    return filtered;
+  }
+
+  private String serializeMetrics(Map<String, Serializable> props, String base) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("trace-id: ");
+    sb.append(props.get(QM_TRACEID));
+    List<String> sortedKeys =
+        props.keySet().stream()
+            .filter(e -> !e.equals(QM_TRACEID))
+            .sorted()
+            .collect(Collectors.toList());
+    for (String key : sortedKeys) {
+      sb.append(", ");
+      sb.append(key);
+      sb.append(": ");
+      sb.append(props.get(key));
+    }
+    return sb.toString();
+  }
+
+  private QueryRequest addTraceId(QueryRequest queryRequest) {
+    String traceId = ThreadContextProperties.getTraceId();
+    queryRequest.getProperties().put(QM_TRACEID, traceId);
+    return queryRequest;
+  }
+
+  private String getMetric(String base, Object o, String suffix) {
+    return base + o.getClass().getSimpleName() + suffix;
+  }
+
+  private QueryResponse putMetricsDuration(QueryResponse response, String key, long value) {
+    putMetricsDuration(response.getRequest(), key, value);
```

Comment:
They are actually being added to both - whichever is handy. Initially, we don't have a response object when collecting pre-query metrics for all the plugins and blocks that happen before the actual query so we store them in the request object. This allows things like the trace-id to be propagated through thread contexts. Once we have a response object, the metrics collected in the post-query steps are captured there. (The request object is maintained within the response object). When reporting the metrics, they are both combined to a single collection before reporting. There's a couple places where we copy things between each since new response objects are generated and included properties would get dropped without explicit copying over (TimedSource).